sdk/core/azure-core/test/ut/md5.cpp

```diff
@@ -62,13 +62,17 @@ uint64_t RandomInt(uint64_t minNumber, uint64_t maxNumber)
   return distribution(random_generator);
 }
 
-TEST(Md5, Basic)
+TEST(Md5Hash, Basic)
```

Comment:
Can you add a test where we create a Hash and never call append or final from it and then we dispose it?

Reply:
What are we testing/asserting?
Are you suggesting just something like this, with nested scope?
```C++
{
  Md5Hash m;
}
```

Reply:
We would be testing the constructor and destructor when no function is called because we have a `delete` statement in the destructor and some casting to the `void *` .
Have you considered how possible would it be to replace the void* for a unique_ptr so we can avoid calling delete?

new and delete are typically meant to bring bugs to the code at some point :)

Reply:
Almost all the logic was borrowed from the existing implementation in storage, so I didn't consider reviewing and changing the implementation too much. 

Correct me if I am mistaken, but I don't think we are allowed to have a void `unique_ptr`.

From what I gather, the use of `void*` and casting is to avoid leaking implementation-only types in the public header.